File: app/src/main/java/io/github/zwieback/familyfinance/business/chart/service/formatter/WeekValueFormatter.java
package io.github.zwieback.familyfinance.business.chart.service.formatter;

import com.github.mikephil.charting.formatter.ValueFormatter;

import org.threeten.bp.LocalDate;

import io.github.zwieback.familyfinance.util.DateUtils;

public class WeekValueFormatter extends ValueFormatter {

    @Override
    public String getFormattedValue(float weeksFromEpoch) {
        LocalDate month = determineCorrectWeek(weeksFromEpoch);
        return DateUtils.localDateToString(month, DateUtils.ISO_LOCAL_WEEK);
    }

    private static LocalDate determineCorrectWeek(float weeksFromEpoch) {
        return DateUtils.epochWeekToLocalDate((long) weeksFromEpoch);
    }
}

File: app/src/main/java/io/github/zwieback/familyfinance/business/chart/service/formatter/WeekValueFormatter.kt
package io.github.zwieback.familyfinance.business.chart.service.formatter

import io.github.zwieback.familyfinance.constant.DateConstants
import io.github.zwieback.familyfinance.extension.WeeksFromEpoch
import org.threeten.bp.format.DateTimeFormatter

class WeekValueFormatter : SpecificPeriodFormatter() {

    override fun getFormatter(): DateTimeFormatter = DateConstants.ISO_LOCAL_WEEK

    override fun getSpecificPeriod(value: Long) = WeeksFromEpoch(value)
}
